docs/guides/deployment/heroku.rst

```diff
@@ -0,0 +1,99 @@
+.. _ref_guide_deployment_heroku:
+
+=========
+On Heroku
+=========
+
+In this guide we show how to deploy EdgeDB to Heroku using a Heroku PostgreSQL
+add-on as the backend.
+
+Because of Heroku's architecture EdgeDB must be deployed with a web app on
+Heroku. For this guide we will use a `todo app written in node <todo-repo_>`_.
+
+.. _todo-repo: https://github.com/edgedb/simpletodo/tree/main
+
+First copy the code, initialize a new git repo, and create a new heroku app.
+
+.. code-block:: bash
+
+   $ npx tiged 'edgedb/simpletodo#main' simpletodo-heroku
+   $ cd simpletodo-heroku
+   $ git init --initial-branch main
+   $ heroku apps:create --buildpack heroku/nodejs
+
+If you are using the `JS query builder for EdgeDB <js-query-builder>`_ then you
+will need to check the ``dbschema/edgeql-js`` directory in to your git repo
+after running ``yarn edgeql-js``. The ``edgeql-js`` command can not be run
+during the build step on Heroku because it needs access to a running EdgeDB
+instance which is not available at build time on Heroku.
+
+.. _js-query-builder: http://localhost:3000/docs/clients/01_js/index
+
+.. code-block:: bash
+
+   $ yarn install && yarn edgeql-js --target cjs
```

Comment:
The `yarn edgeql-js` requires `edgedb project init` first.

Should probably be `npx edgeql-js`, even Yarn users use `npx` for stuff like this.

You need to run `yarn build` before `yarn start`, otherwise you'll get an error like this:

```
➜  simpletodo-heroku git:(main) ✗ yarn start
yarn run v1.22.5
$ next start -p ${PORT:=3000}
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Using webpack 5. Reason: no custom webpack configuration in next.config.js https://nextjs.org/docs/messages/webpack5
Error: Could not find a production build in the '/Users/colinmcd94/Documents/edgedb/playgrounds/scratch/simpletodo-heroku/.next' directory. Try building your app with 'next build' before starting the production server. https://nextjs.org/docs/messages/production-start-no-build-id
    at Server.readBuildId (/Users/colinmcd94/Documents/edgedb/playgrounds/scratch/simpletodo-heroku/node_modules/next/dist/next-server/server/next-server.js:151:355)
    at new Server (/Users/colinmcd94/Documents/edgedb/playgrounds/scratch/simpletodo-heroku/node_modules/next/dist/next-server/server/next-server.js:3:120)
    at NextServer.createServer (/Users/colinmcd94/Documents/edgedb/playgrounds/scratch/simpletodo-heroku/node_modules/next/dist/server/next.js:1:2935)
    at async /Users/colinmcd94/Documents/edgedb/playgrounds/scratch/simpletodo-heroku/node_modules/next/dist/server/next.js:1:3360
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
> You need to run yarn build before yarn start

Heroku runs `yarn build` before deploying the app. Why do we need to run `yarn start` locally? Isn't `yarn dev` more appropriate for running locally?